src/strategies/xdai-stake/index.ts

```diff
@@ -0,0 +1,133 @@
+import { subgraphRequest, call } from '../../utils';
+import { getBlock } from '../../utils/web3';
+import { strategy as erc20BalanceOfStrategy } from '../erc20-balance-of';
+import { calculateEmission } from './utils';
+
+export const author = 'maxaleks';
+export const version = '0.1.0';
+
+const  EASY_STAKING_SUBGRAPH_URL = {
+  '1': 'https://api.thegraph.com/subgraphs/name/maxaleks/easy-staking',
+};
+
+const ercABI = [
+  {
+		constant: true,
+		inputs: [],
+		name: 'totalSupply',
+		outputs: [
+			{
+				name: '',
+				type: 'uint256'
+			}
+		],
+		payable: false,
+		stateMutability: 'view',
+		type: 'function'
+	}
+];
+
+async function getEasyStakingDeposits(network, addresses, snapshot) {
+  const params = {
+    deposits: {
+      __args: {
+        where: {
+          user_in: addresses.map((address) => address.toLowerCase()),
+          amount_gt: 0,
+        },
+        first: 1000,
+        skip: 0,
+      },
+      user: true,
+      amount: true,
+      timestamp: true,
+    },
+  };
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.deposits.__args.block = { number: snapshot };
+  }
+  let page = 0;
+  let deposits = [];
+  while(true) {
+    params.deposits.__args.skip = page * 1000;
+    const data = await subgraphRequest(EASY_STAKING_SUBGRAPH_URL[network], params);
+    deposits = deposits.concat(data.deposits);
+    page++;
+    if (data.deposits.length < 1000) break;
+  }
+  return deposits.map(deposit => ({
+    ...deposit,
+    amount: BigNumber.from(deposit.amount),
+    timestamp: BigNumber.from(deposit.timestamp),
+  }));
+}
+
+async function getEasyStakingParams(network, snapshot) {
+  const params = {
+    commonData: {
+      __args: {
+        id: 'common',
+      },
+      sigmoidParamA: true,
+      sigmoidParamB: true,
+      sigmoidParamC: true,
+      totalSupplyFactor: true,
+      totalStaked: true,
+    },
+  };
+  if (snapshot !== 'latest') {
+    // @ts-ignore
+    params.commonData.__args.block = { number: snapshot };
+  }
+  const { commonData } = await subgraphRequest(EASY_STAKING_SUBGRAPH_URL[network], params);
+  return {
+    sigmoidParameters: {
+      a: BigNumber.from(commonData.sigmoidParamA),
+      b: BigNumber.from(commonData.sigmoidParamB),
+      c: BigNumber.from(commonData.sigmoidParamC),
+    },
+    totalSupplyFactor: BigNumber.from(commonData.totalSupplyFactor),
+    totalStaked: BigNumber.from(commonData.totalStaked),
+  };
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const [
+    easyStakingDeposits,
+    { sigmoidParameters, totalSupplyFactor, totalStaked },
+    erc20Score,
+    block,
+    totalSupply,
+  ] = await Promise.all([
+    getEasyStakingDeposits(network, addresses, snapshot),
+    getEasyStakingParams(network, snapshot),
+    erc20BalanceOfStrategy(space, network, provider, addresses, options, snapshot),
+    getBlock(provider, snapshot),
```

Comment:
You can use `provider.getBlock(snapshot)` directly the new function `getBlock` is not necessary.

Reply:
You're right, sorry 😅